Zend/zend_API.h

```diff
@@ -1162,8 +1235,18 @@ static zend_always_inline int zend_parse_arg_str(zval *arg, zend_string **dest,
 	if (EXPECTED(Z_TYPE_P(arg) == IS_STRING)) {
 		*dest = Z_STR_P(arg);
 	} else if (EXPECTED(Z_TYPE_P(arg) < IS_STRING)) {
-		if (check_null && UNEXPECTED(Z_TYPE_P(arg) == IS_NULL)) {
-			*dest = NULL;
+		if (UNEXPECTED(Z_TYPE_P(arg) == IS_NULL)) {
+			if (check_null && (Z_TYPE_P(arg) == IS_NULL)) {
+				*dest = NULL;
+#if STH_DISABLE_NULL_TO_STRING	
+			} else {
+				return 0;
+#endif
```

Comment:
you need an else here to prevent a segfault (null pointer). I added a goto to the conversion logic in an else branch and it worked fine:

@@ -1241,6 +1241,9 @@ static zend_always_inline int zend_parse_arg_str(zval *arg, zend_string **dest,
 #if STH_DISABLE_NULL_TO_STRING 
                        } else {
                                return 0;
+#else
+            } else {
+                goto convert;        
 #endif
                        }
 #if STH_DISABLE_BOOL_TO_STRING
@@ -1248,6 +1251,7 @@ static zend_always_inline int zend_parse_arg_str(zval *arg, zend_string **dest,
                        return 0;
 #endif
                } else {
+convert:
                        if (Z_COPYABLE_P(arg) && Z_REFCOUNT_P(arg) > 1) {